src/server/debug/server/server.go

```diff
@@ -158,7 +162,10 @@ func (s *debugServer) GetDumpV2Template(ctx context.Context, request *debug.GetD
 	}, nil
 }
 
-func (s *debugServer) listApps(ctx context.Context) (_ []*debug.App, retErr error) {
+// list apps returns a list of running apps, and a list of apps which may possibly exist.  The
+// intent is to use the first result for things like "kubectl describe" and the second for getting
+// loki logs.
```

Comment:
Could this be refactored? Might it be clearer to merge `podApps` (which would be the old `listApps` and `pipelineApps` (which would generate all apps for all known pipelines), rather than in `GetDumpV2Template` listing pipelines, passing that into `listApps`, then in `listApps` iterating over running pods, deleting a pipeline pod’s associated pod from the list of pipelines, then creating an app for each remaining pipeline?

Reply:
This should really be implemented in some completely different way; running pods should only affect logs, loki should be queried to enumerate all logs that loki can get, etc.  But this is a release blocker because everyone is using autoscaling now and all the debug dumps that people are generating are useless, so I'm trying to keep it as minimal as possible for the backport.